internal/xds/translator/jsonpatch.go

```diff
@@ -84,9 +84,9 @@ func processJSONPatches(tCtx *types.ResourceVersionTable, envoyPatchPolicies []*
 				}
 			}
 
-			// If Path is "" and op is "add", unmarshal and add the patch as a complete
+			// If Path and JSONPath is "" and op is "add", unmarshal and add the patch as a complete
 			// resource
-			if p.Operation.Op == AddOperation && p.Operation.Path == EmptyPath {
+			if p.Operation.Op == AddOperation && p.Operation.IsPathEmpty() && p.Operation.IsJSONPathEmpty() {
```

Comment:
amd when testing for IsEmpty - ` o.Path != nil && *o.Path == EmptyPath`

Reply:
Agree, will change this, then its more explicit.